lib/ios-log.js

```diff
@@ -0,0 +1,217 @@
+      this.finishStartingLogCapture();
+      return;
+    }
+
+    let logsPath;
+
+    if (ver >= 6) { 
+      logger.debug('Starting iOS 8.* simulator log capture');
+        if (_.isUndefined(this.simUdid)) {
+          logger.errorAndThrow('ios8 log capture requires a sim udid'); 
+        }
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+    } else {
+        logger.debug('Starting iOS7.* simulator log capture');
+        logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+    }
+
+    try {
+      if (logsPath.indexOf('*') >= 0) {
+        return; 
+      }
+        await mkdirp(logsPath);
+        await touch(path.resolve(logsPath, 'system.log'));
+        await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+              'A new Simulator is about to start!'); 
+        let files;
+        try {
+          files = await glob(path.resolve(logsPath, 'system.log'));
+        if (files.length < 1) {
+          throw 'Could not start log capture';
+        }
+      } catch (e) {
+        logger.error('Could not start log capture because no iOS ' +
+        'simulator logs could be found at ' + logsPath + '/system.log' +
+        'Logging will not be functional for this run');
+      }
+
+      let lastModifiedLogPath = files[0];
+      let lastModifiedLogTime = await fs.stat(files[0]).mtime; //promisify this 
+      for (let file in files) {
+        let mtime = await fs.stat(file).mtime;
+        if (mtime > lastModifiedLogTime) {
+          lastModifiedLogPath = file;
+          lastModifiedLogTime = mtime;
+        }
+      }
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+      this.finishStartingLogCapture();
+    } catch (err) {
+      logger.warn('System log capture failed.');
+      return;
+    }    
+  }  
+
+  //set up the procedure with stdout stderr, once it has any output -- > resolve the promise, show logs have started
+  // 
+
+  // first time you want to ignore it, 
+  // have the promise use a local varialbe outside of promise, and change the boolean after the first line is done. 
+  // make first lien empty 
+  finishStartingLogCapture () {
+    if (!this.proc) {
+      let msg = 'Could not capture device log';
+      logger.warn(msg);
+      throw new Error(msg);
+    }
+     this.startTimer = setTimeout(function () {
+      let msg = 'Log capture did not start in a reasonable amount of time';
+      logger.error(msg);
+
+      throw new Error(msg);
+      
+    }, START_TIMEOUT);
+    let firstLine = true;
+    return new Promise((resolve, reject) => {
+      this.proc.on('output', (stdout, stderr) => {
+        if (stdout) {
+          //log.info(`[STDOUT] ${stdout.trim()}`);
+          clearTimeout(this.startTimer);
+          if (firstLine) {
+            this.onOutput(stdout);
+          } else {
+            this.logRow += stdout;
```

Comment:
Will this code be ever called because i think `firstLine` would never be set to `false`, am i missing something?